### PR TITLE
Reload the config file before applying a profile

### PR DESCRIPTION
**Pull request by ghost**
I've been toying with saving a kanshi config file from [wdisplays](https://github.com/cyclopsian/wdisplays). Whenever the user changes some display settings, it should rewrite the kanshi config (without clobbering any inactive profiles) and then apply the settings via wlr_output_manager. Kanshi then needs to reload the config file otherwise it will immediately try to reset the outputs back to their previous configuration. This PR was the simplest way to get it to work without having to manually restart the daemon, but I'm not sure this is the right way to go about it. Maybe it should stat and check the modified time of the config file before re-parsing it? I didn't want to rely on any external programs to restart kanshi either. For reference, the necessary changes to wdisplays are here: https://github.com/cyclopsian/wdisplays/pull/5

### CODE REVIEW

File: parser.c [hunk 1]
```diff
@@ -33,6 +33,7 @@ static int parser_read_char(struct kanshi_parser *parser) {
 		return ch;
 	}
 
+	errno = 0;
```
Comment (ghost) [inline parser.c:36]: I needed to do this because sometimes fgetc doesn't seem to reset errno, not sure what causes it.

**Comment by emersion** [comment]
Thanks for your PR! However I'm wondering whether it's the way to go or not...

What happens if the user edits the current profile?

I wonder if we want to do implicit config reload at all. We're just lucky that for kanshi we have an event when we can figure out whether we need to reload (but doesn't work in all cases, see above). Related: https://github.com/emersion/kanshi/issues/13

**Comment by ghost** [comment]
@emersion Yes, nothing happens if the user edits the config, unless the user knows enough to manually trigger some output changes (unplug and replug the monitor, run swaymsg output, etc). This PR is only the bare minimum for other programs to reconfigure kanshi.

There was another branch I was working on that has an event loop and a really rough IPC implementation over a named pipe, see here: https://github.com/emersion/kanshi/compare/master...cyclopsian:kanshictl

I didn't submit that branch though because it seemed very hacky as-is. It could be used as a base to drop in varlink instead of the named pipe. But that also seems kind of like overkill just to support one IPC message with no return value and no parameters. Let me know what you think.

**Comment by emersion** [comment]
>I didn't submit that branch though because it seemed very hacky as-is. It could be used as a base to drop in varlink instead of the named pipe. But that also seems kind of like overkill just to support one IPC message with no return value and no parameters. Let me know what you think.

We have other use-cases for IPC, see e.g. https://github.com/emersion/kanshi/issues/10

But I'd be fine with handling a signal to reload the config, too.

**Comment by ghost** [comment]
@emersion Ah okay, that can all be done in one go then. I will close this PR and take a look at integrating varlink, that sounds like a useful way to go about things. It should also be trivial to add a SIGHUP handler.

**Comment by ammen99** [comment]
Maybe a bit too late, but why not have kanshi add an inotify watch on the config file?

**Comment by emersion** [comment]
This is not portable, and I don't want to reload automatically when the user saves the file, I'd prefer the reload to be explicit (so that users can write `output * disable` by mistake without ending up with an unusable session).

**Pull request closed without merging**